apps/mobile/src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,69 @@
+import { useEffect } from 'react'
+import FCMService from '@/src/services/notifications/FCMService'
+import { useAppSelector, useAppDispatch } from '@/src/store/hooks'
+import {
+  selectAppNotificationStatus,
+  selectDeviceNotificationStatus,
+  toggleAppNotifications,
+} from '@/src/store/notificationsSlice'
+import { STORAGE_IDS } from '@/src/store/constants'
+import { reduxStorage } from '@/src/store/storage'
+import NotificationService from '@/src/services/notifications/NotificationService'
+
+const useNotifications = () => {
+  const dispatch = useAppDispatch()
+  /**
+   * We need to check if the user has enabled notifications for the device in order to keep listening for messages
+   * since the user can disable notifications at any time on their device, we need to handle app behavior accordingly
+   * if device notifications are disabled, the user has been prompt more than 3 times within a month to enable the app notifications
+   * we should only ask the user to enable notifications again after a month has passed
+   *
+   * If the user has disabled notifications for the app,  we should disable app notifications
+   */
+  const isDeviceNotificationEnabled = useAppSelector(selectDeviceNotificationStatus)
+  const isAppNotificationEnabled = useAppSelector(selectAppNotificationStatus)
+
+  useEffect(() => {
+    const checkNotifications = async () => {
+      if (!isDeviceNotificationEnabled) {
+        const promptAttempts = reduxStorage.getItem(STORAGE_IDS.PUSH_NOTIFICATIONS_PROMPT_COUNT)
```

Comment:
I'm confused. Why do we have to do reduxStorage.getItem instead of using a selector as we did above for isDeviceNotificationEnabled?